lib/cartopy/util.py

```diff
@@ -91,3 +90,313 @@ def add_cyclic_point(data, coord=None, axis=-1):
+     [0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]]
+    >>> print(cyclic_lon2d)
+    [[  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]]
+
+    Adding a cyclic point to a data array and the associated 2-dimensional
+    x- and y-coordinates.
+    >>> lons = np.arange(0, 360, 60)
+    >>> lats = np.arange(-90, 90, 180/5)
+    >>> lon2d, lat2d = np.meshgrid(lons, lats)
+    >>> cyclic_data, cyclic_lon2d, cyclic_lat2d = add_cyclic(
+    ...     data, x=lon2d, y=lat2d)
+    >>> print(cyclic_data)  # doctest: +NORMALIZE_WHITESPACE
+    [[0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]
+     [0. 1. 2. 3. 4. 5. 0.]]
+    >>> print(cyclic_lon2d)
+    [[  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]]
+    >>> print(cyclic_lat2d)
+    [[-90. -90. -90. -90. -90. -90. -90.]
+     [-54. -54. -54. -54. -54. -54. -54.]
+     [-18. -18. -18. -18. -18. -18. -18.]
+     [ 18.  18.  18.  18.  18.  18.  18.]
+     [ 54.  54.  54.  54.  54.  54.  54.]]
+
+    Not adding a cyclic point if cyclic point detected in x.
+    >>> lons = np.arange(0, 361, 72)
+    >>> lats = np.arange(-90, 90, 180/5)
+    >>> lon2d, lat2d = np.meshgrid(lons, lats)
+    >>> cyclic_data, cyclic_lon2d, cyclic_lat2d = add_cyclic(
+    ...     data, x=lon2d, y=lat2d)
+    >>> print(cyclic_data)  # doctest: +NORMALIZE_WHITESPACE
+    [[0. 1. 2. 3. 4. 5.]
+     [0. 1. 2. 3. 4. 5.]
+     [0. 1. 2. 3. 4. 5.]
+     [0. 1. 2. 3. 4. 5.]
+     [0. 1. 2. 3. 4. 5.]]
+    >>> print(cyclic_lon2d)
+    [[  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]]
+    >>> print(cyclic_lat2d)
+    [[-90. -90. -90. -90. -90. -90.]
+     [-54. -54. -54. -54. -54. -54.]
+     [-18. -18. -18. -18. -18. -18.]
+     [ 18.  18.  18.  18.  18.  18.]
+     [ 54.  54.  54.  54.  54.  54.]]
+
+    """
+    if x is None:
+        return _add_cyclic_data(data, axis=axis)
+    # if x was given
+    if x.ndim > 2:
+        xaxis = axis
+    else:
+        xaxis = -1
+    if x.shape[xaxis] != data.shape[axis]:
+        estr = (f'x.shape[{xaxis}] does not match the size of the'
+                f' corresponding dimension of the data array:'
+                f' x.shape[{xaxis}] = {x.shape[xaxis]},'
+                f' data.shape[{axis}] = {data.shape[axis]}.')
+        raise ValueError(estr)
+    if has_cyclic(x, axis=xaxis, cyclic=cyclic, precision=precision):
+        if y is None:
+            return data, x
+        # if y was given
+        return data, x, y
+    # if not has_cyclic, add cyclic points to data and x
+    odata = _add_cyclic_data(data, axis=axis)
```

Comment:
```suggestion
    out_data = _add_cyclic_data(data, axis=axis)
```
I think you meant out here? I'd be explicit in the naming because it isn't very long. Same for the other "o"s